src/Message/JsonMessageSerializer.php

```diff
@@ -18,6 +18,7 @@ public function serialize(MessageInterface $message): string
             'name' => $message->getHandlerName(),
             'data' => $message->getData(),
             'meta' => $message->getMetadata(),
+            'class' => $message instanceof EnvelopeInterface ? $message->getMessage()::class : $message::class,
```

Comment:
Am I correct that class name is in the data that is sent? If so, I'd avoid that. Is that possible to use a plain envelope name, not a class?

Reply:
It's a message class. It may be used only by an application which pushed this message to the queue. I'll move this data to the "meta" key. Anyway this data is not usable by another services.